### PR TITLE
sinkv2(ticdc): fix update resolved ts not taking effect issue

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?

- delete buffer after flush
- set correct eventID
- add more tests

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sinkv2/tablesink/event_table_sink.go [hunk 1]
```diff
@@ -58,18 +74,22 @@ func (e *eventTableSink[E]) UpdateResolvedTs(resolvedTs model.ResolvedTs) {
 		return
 	}
 	resolvedEvents := e.eventBuffer[:i]
+	e.eventBuffer = append(make([]E, 0, len(e.eventBuffer[i:])), e.eventBuffer[i:]...)
```
Comment (asddongmen) [inline cdc/sinkv2/tablesink/event_table_sink.go:77]: How about add a comment to explain why do you to reset the `eventBuffer` in this way.

**Review by asddongmen — APPROVED** [review]

File: cdc/sinkv2/eventsink/event.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ type CallbackFunc func()
 type CallbackableEvent[E TableEvent] struct {
 	Event       E
 	Callback    CallbackFunc
-	TableStatus *pipeline.TableState
+	TableStatus pipeline.TableState
```
Comment (overvenus) [inline cdc/sinkv2/eventsink/event.go:35]: Why need `TableStatus` and why change to value?

Comment (hi-rustin) [reply cdc/sinkv2/eventsink/event.go]: It's a mistake. Fixed and added test for it.

**Review by hicqu — APPROVED** [review]

**Pull request merged**